metaflow/plugins/cards/card_decorator.py

```diff
@@ -0,0 +1,192 @@
+        # Py2 raises ValueError
+        # https://stackoverflow.com/questions/53355389/python-2-3-compatibility-issue-with-exception
+        try:
+            import json
+
+            RaisingError = json.decoder.JSONDecodeError
+        except AttributeError:  # Python 2
+            RaisingError = ValueError
+
+        self.card_options = None
+
+        # Populate the defaults which may be missing.
+        missing_keys = set(self.defaults.keys()) - set(self.attributes.keys())
+        for k in missing_keys:
+            self.attributes[k] = self.defaults[k]
+
+        if type(self.attributes["options"]) is str:
+            try:
+                self.card_options = json.loads(self.attributes["options"])
+            except RaisingError:
+                self.card_options = self.defaults["options"]
+        else:
+            self.card_options = self.attributes["options"]
+
+        self._flow_datastore = flow_datastore
+        self._environment = environment
+
+    def task_pre_step(
+        self,
+        step_name,
+        task_datastore,
+        metadata,
+        run_id,
+        task_id,
+        flow,
+        graph,
+        retry_count,
+        max_user_code_retries,
+        ubf_context,
+        inputs,
+    ):
+        self._task_datastore = task_datastore
+        self._metadata = metadata
+
+    def task_finished(
+        self, step_name, flow, graph, is_task_ok, retry_count, max_user_code_retries
+    ):
+        if not is_task_ok:
+            return
+        runspec = "/".join([current.run_id, current.step_name, current.task_id])
+        self._run_cards_subprocess(runspec)
+
+    @staticmethod
+    def _options(mapping):
+        for k, v in mapping.items():
+            if v:
+                k = k.replace("_", "-")
+                v = v if isinstance(v, (list, tuple, set)) else [v]
+                for value in v:
+                    yield "--%s" % k
+                    if not isinstance(value, bool):
+                        yield to_unicode(value)
+
+    def _create_top_level_args(self):
+
+        top_level_options = {
+            "quiet": True,
+            "metadata": self._metadata.TYPE,
+            "coverage": "coverage" in sys.modules,
+            "environment": self._environment.TYPE,
+            "datastore": self._flow_datastore.TYPE,
+            "datastore-root": self._flow_datastore.datastore_root,
+            # We don't provide --with as all execution is taking place in
+            # the context of the main processs
+        }
+        # FlowDecorators can define their own top-level options. They are
+        # responsible for adding their own top-level options and values through
+        # the get_top_level_options() hook.
+        for deco in flow_decorators():
+            top_level_options.update(deco.get_top_level_options())
+        return list(self._options(top_level_options))
+
+    def _run_cards_subprocess(self, runspec):
```

Comment:
Quick note here; I don't think it matters much but in a local mode this will effectively double the number of processes which may not be desirable. I don't think it is a huge issue but pointing it out.